### PR TITLE
chore(package): update eslint to version 3.9.1

### PR DESCRIPTION
**Pull request by nguyenchr**
@srushti @rprieto @kiddkai @vickvu 

now that we no longer support node 0.10, this PR should work

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -2,25 +2,17 @@ env:
...
-  wrap-iife: 2
-  quotes: "single"
-  strict: false
-  new-cap: false
-  no-multi-spaces: false
-  curly: true
-  guard-for-in: false
-  no-underscore-dangle: false
-  no-new: false
-  no-wrap-func: false
```
Comment (srushti) [inline .eslintrc]: Are all the rules above carried over below? Looks like there's way fewer lines. Are they all included in `eslint:recommended`?

File: test/matchers/oneof.spec.js [hunk 2]
```diff
@@ -30,6 +30,7 @@ describe('oneOf', function() {
   });
 
   it('will have errors when not matching any of the schemas', function() {
+    console.log(oneOf(['string', 'number']).match({}))
```
Comment (srushti) [inline test/matchers/oneof.spec.js]: Was this just meant to be for when you were debugging this test?

Comment (nguyenchr) [inline test/matchers/oneof.spec.js]: good pickup

**Comment by nguyenchr** [comment]
@srushti updated the eslintrc file and removed any rules that were set to `false`

File: .eslintrc [hunk 3]
```diff
@@ -10,17 +10,10 @@ rules:
   no-nested-ternary: 2
   no-undefined: 2
   radix: 2
-  space-after-keywords: [2, "always"]
+  keyword-spacing: [2, { "before": true, "after": true}]
```
Comment (srushti) [inline .eslintrc]: Not that it's a big deal but does it need to be `"before":` rather than just `before`?

Comment (nguyenchr) [inline .eslintrc]: not sure if it has to be but that's how it's specified in the docs

**Pull request merged**